export/export_pipeline/compute_collision_representation.py

```diff
@@ -0,0 +1,427 @@
+                shutil.copy(mesh_path, out_filename)
+            else:
+                mesh.export(out_filename)
+            decomposed = True
+        elif acd_args['decompose_method'] == 'skip':
+            decomposed = False
+        else:
+            raise ValueError(f'{mesh_dir} decompose method {acd_args["decompose_method"]} is invalid')
+        
+        print(f"Removing {mesh_path}")
+        os.remove(mesh_path)
+        
+    if not decomposed:  # delete whole directory
+        shutil.rmtree(mesh_dir)
+    return decomposed
+
+
+class CollisionRepresentationComputer(object):
+    def __init__(self, scene_path: str, n_workers: int, rerun: bool, include_objects: str) -> None:
+        self.n_workers = min(max(1, n_workers), mp.cpu_count()-1)
+        self.rerun = rerun
+        self.scene_path = osp.normpath(scene_path)
+        if include_objects is not None:
+            include_objects = include_objects.split(',')
+        self.input_gltf_filenames = []
+        with open(osp.join('export_pipeline', 'params.yaml'), 'r') as f:
+            self.p = yaml.safe_load(f)
+        self.input_dir = osp.join(scene_path, self.p['common']['GLTF_SCENE_DIR_NAME'])
+        self.output_dir = osp.join(scene_path, self.p['common']['COLLISION_DIR_NAME'])
+        for filename in next(os.walk(self.input_dir))[2]:
+            if not filename.endswith('.gltf'):
+                continue
+            if include_objects and not any([osp.splitext(filename)[0] == i for i in include_objects]):
+                continue
+            self.input_gltf_filenames.append(osp.join(self.input_dir, filename))
+
+
+    def run(self):
+        if not self.rerun:
+            shutil.rmtree(self.output_dir, ignore_errors=True)
+
+        bpy.ops.object.select_all(action='SELECT')
+        bpy.ops.object.delete()
+        
+        with open(osp.join(self.input_dir, 'actors_information.json'), 'r') as f:
+            actors_information = json.load(f)
+        
+        process_mesh_args = []
+        for input_gltf_filename in self.input_gltf_filenames:
+            bpy.ops.import_scene.gltf(filepath=input_gltf_filename)
+            obj = bpy.context.object
+            if obj.parent is not None:
+                continue
+            
+            object_name = obj.name.split('.')[0]
+            obj_dir = os.path.join(self.output_dir, object_name)
+            obj_info = actors_information[object_name]
+
+            joint_filename = input_gltf_filename.replace('.gltf', '_joints.json')
+            try:
+                with open(joint_filename, 'r') as f:
+                    joints_info = json.load(f)
+                joints_info = {j['name']: j for j in joints_info}
+            except FileNotFoundError:
+                joints_info = {}
+            
+            # traverse the tree, collect components info
+            geoms = []
+            components_info = {'static': {}, 'joints': {}}
```

Comment:
I agree with having top-level `static` and `joints` sections in our json file. But in our code, we don't need to create nested dictionaries that match our final json representation exactly when we are still manipulating and populating the dictionaries. It would be more readable if we had `static_mesh_components = {}` and `joint_components = {}` or something, and serialized them to json when we're finished manipulating them.